cMake/FreeCAD_Helpers/CompilerChecksAndSetups.cmake

```diff
@@ -9,19 +9,7 @@ macro(CompilerChecksAndSetups)
 
     # ================================================================================
 
-    # Needed for boost1.69
-    # Avoid that Python (pyerrors.h) defines snprintf and vsnprintf
-    if (MSVC AND NOT MSVC_VERSION VERSION_LESS 1900)
-        add_definitions(-DHAVE_SNPRINTF)
-    elseif (MINGW)
-        add_definitions(-DHAVE_SNPRINTF)
-    endif()
-
-    # Allow developers to use Boost < 1.55
```

Comment:
Please don't change too many different things. This PR should be to address CMake.
Please make a separate PR for boost.